scripts/srt-dev.lua

```diff
@@ -0,0 +1,887 @@
+-- @brief srt-dev Protocol dissector plugin
+
+-- create a new dissector
+local NAME = "SRT-dev"
+local srt_dev = Proto(NAME, "SRT-dev Protocol")
+
+-- create a preference of a Protocol
+srt_dev.prefs["srt_udp_port"] = Pref.uint("SRT UDP Port", 1935, "SRT UDP Port")
+
+-- create fields of srt_dev
+-- Base.HEX, Base.DEC, Base.OCT, Base.UNIT_STRING, Base.NONE
+local fields = srt_dev.fields
+-- General field
+local pack_type_select = {
+	[0] = "Data Packet",
+	[1] = "Control Packet"
+}
+fields.pack_type_tree = ProtoField.uint32(NAME .. ".pack_type_tree", "Packet Type", base.HEX)
+fields.pack_type = ProtoField.uint16("srt_dev.pack_type", "Packet Type", base.HEX, pack_type_select, 0x8000)
+fields.reserve = ProtoField.uint16("srt_dev.reserve", "Reserve", base.DEC)
+fields.additional_info = ProtoField.uint32("srt_dev.additional_info", "Additional Information", base.DEC)
+fields.time_stamp = ProtoField.uint32("srt_dev.time_stamp", "Time Stamp", base.DEC)
+fields.dst_sock = ProtoField.uint32("srt_dev.dst_sock", "Destination Socket ID", base.DEC)
+fields.none = ProtoField.none("srt_dev.none", "none", base.NONE)
+
+-- Data packet fileds
+fields.data_flag_info_tree = ProtoField.uint8("srt_dev.data_flag_info_tree", "Data Flag Info", base.HEX)
+local FF_state_select = {
+	[0] = "[Middle packet]",
+	[1] = "[Last packet]",
+	[2] = "[First packet]",
+	[3] = "[Single packet]"
+}
+fields.FF_state = ProtoField.uint8("srt_dev.FF_state", "FF state", base.HEX, FF_state_select, 0xC0)
+local O_state_select = {
+	[0] = "[Data delivered unordered]",
+	[1] = "[Data delivered in order]"
+}
+fields.O_state = ProtoField.uint8("srt_dev.O_state", "O state", base.HEX, O_state_select, 0x20)
+local KK_state_select = {
+	[0] = "[Not encrypted]",
+	[1] = "[Data encrypted with even key]",
+	[2] = "[Data encrypted with odd key]"
+}
+fields.KK_state = ProtoField.uint8("srt_dev.KK_state", "KK state", base.HEX, KK_state_select, 0x18)
+local R_state_select = {
+	[0] = "[Not Set]",
+	[1] = "[Retransmit Packet]"
```

Comment:
This bit distinguished regularly (normal, first time) sent packet to a retransmitted packet, so the most appropriate names here are "ORIGINAL" (when clear) and "RETRANSMITTED" (when set).